docs/content/integrations/duckdb/reference.mdx

```diff
@@ -7,17 +7,79 @@ description: Store your Dagster assets in DuckDB
 
+DuckDB resource:
+
+- [Using the DuckDB resource in assets](#using-the-duckdb-resource-in-assets)
+- [Using the DuckDB resource in ops](#using-the-duckdb-resource-in-ops)
+
+DuckDB I/O manager:
+
 - [Selecting specific columns in a downstream asset](#selecting-specific-columns-in-a-downstream-asset)
 - [Storing partitioned assets](#storing-partitioned-assets)
 - [Storing tables in multiple schemas](#storing-tables-in-multiple-schemas)
 - [Using the DuckDB I/O manager with other I/O managers](#using-the-duckdb-io-manager-with-other-io-managers)
 - [Storing and loading PySpark or Polars DataFrames in DuckDB](#storing-and-loading-pyspark-or-polars-dataframes-in-duckdb)
 - [Storing multiple DataFrame types in DuckDB](#storing-multiple-dataframe-types-in-duckdb)
-- [Executing custom SQL commands with the DuckDB resource](#executing-custom-sql-commands-with-the-duckdb-resource)
+
+
+
+---
+
+## DuckDB resource
+
+The DuckDB resource provides access to a [`duckdb.DuckDBPyConnection`](https://duckdb.org/docs/api/python/reference/#duckdb.DuckDBPyConnection) object. This allows you full control over how your data is stored and retrieved in your database.
+
+For further information on the DuckDB resource, see the [DuckDB resource API docs](/\_apidocs/libraries/dagster-duckdb#dagster_duckdb.DuckDBResource).
+
+### Using the DuckDB resource in assets
+
+```python file=/integrations/duckdb/reference/resource.py startafter=start endbefore=end
+from dagster_duckdb import DuckDBResource
+
+from dagster import asset
+
+# this example executes a query against the iris_dataset table created in Step 2 of the
+# Using Dagster with DuckDB tutorial
+
+
+@asset(deps=[iris_dataset])
+def small_petals(duckdb: DuckDBResource) -> None:
+    with duckdb.get_connection() as conn:  # conn is a DuckDBPyConnection
+        conn.cursor().execute(
+            "CREATE TABLE iris.small_petals AS SELECT * FROM iris.iris_dataset WHERE"
+            " 'petal_length_cm' < 1 AND 'petal_width_cm' < 1"
+        )
+```
+
+In this example, we attach the DuckDB resource to the `small_petals` asset. In the body of the asset function, we use the `get_connection` context manager on the resource to get a [`duckdb.DuckDBPyConnection`](https://duckdb.org/docs/api/python/reference/#duckdb.DuckDBPyConnection). We can use this connection to execute a custom SQL query against the `iris_dataset` table created in [Step 2: Create tables in DuckDB](/integrations/duckdb/using-duckdb-with-dagster#store-a-dagster-asset-as-a-table-in-duckdb) of the [Using Dagster with DuckDB tutorial](/integrations/duckdb/using-duckdb-with-dagster). When the `duckdb.get_connection` context is exited, the created DuckDB connections will be closed.
+
+### Using the DuckDB resource in ops
```

Comment:
how do we feel about this section? there's nothing particularly novel about using the resource in an op vs in an asset, but maybe this would be helpful for the op users out there? maybe it would just add confusion about when to use assets vs ops?

Reply:
I have feelings about this, but I'm not totally sure what those feelings are. Not helpful, I know.

1. **On ops vs assets**: On one hand, I like having an ops-specific example for coverage purposes, but on the other... we're really focusing on assets as the best way to use Dagster. Having this be so high up on the page (and presented as a sorta-binary against assets) feels contradictory to that.
2. **The heading change:** This was the `Executing custom SQL queries` example originally, right? IMO having the example lead with that is really valuable - I know @tacastillo has mentioned that folks ask about how to execute SQL on support not-infrequently. Re-naming this and having the description of what it's doing buried in the explanation may be detrimental.

   Perhaps just adding a sentence before the example itself would be helpful - something like "In this example, we're using the DuckDB resource to execute a custom SQL query"

Reply:
1 - i might just be overindexing on a recent user convo where they were trying to do bigquery resource stuff in ops and got a little confused. an example might have helped them but idk if that's enough justification to add it in 

for 2 - that's really valuable insight and i think a good reason to change the heading back to how it was before

Reply:
I went ahead and removed the ops example and put the heading back to what it was before